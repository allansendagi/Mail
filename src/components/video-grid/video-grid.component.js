import React from 'react';
import './video-grid.styles.scss';
import {VideoGridHeader} from './video-grid-header/video-grid-header.component';
import {Divider} from 'semantic-ui-react';
import {VideoPreview} from '../video-preview/video-preview.component';

export function VideoGrid(props) {
	const divider = props.hideDivider ? null : <Divider/>;

	return (
		<React.Fragment>
		 <div className='video-section'>
		   <VideoGridHeader title='Trending'/>
		    <div className='video-grid'>
		    <div className='video-grid'>
		    <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
	        <VideoPreview/>
		  </div>
		  </div>
		  </div>
		  {divider}
		</React.Fragment>
	);
}
	
