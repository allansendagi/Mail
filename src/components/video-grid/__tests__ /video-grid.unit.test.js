import React from 'react';
import {shallow, configure} from 'enzyme';
import {VideoGrid} from '../video-grid.component';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe('VideoGrid', () => {
  test('renders without props', () => {
    const wrapper = shallow(<VideoGrid/>);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders with title prop', () => {
    const wrapper = shallow(<VideoGrid title='Trending'/>);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders without divider', () => {
    const wrapper = shallow(<VideoGrid hideDivider={true}/>);
    expect(wrapper).toMatchSnapshot();
  });
});