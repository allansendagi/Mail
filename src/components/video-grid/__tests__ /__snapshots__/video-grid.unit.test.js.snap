// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders with title prop 1`] = `ShallowWrapper {}`;

exports[`VideoGrid renders without divider 1`] = `ShallowWrapper {}`;

exports[`VideoGrid renders without props 1`] = `ShallowWrapper {}`;
